src/Npgsql/MultiHostConnectorPool.cs

```diff
@@ -16,7 +16,9 @@ sealed class MultiHostConnectorPool : ConnectorSource
 
         readonly ConnectorSource[] _pools;
```

Comment:
Not really related to this PR, but as I see this code here:

Should the `_pools` array be padded to account for false sharing?
For arrays in .NET the length is stored before the first element, and this length need to be read for bound checks, so when the elements in the same cache line will get updated, false sharing will occur.
See e.g. [a simple demo](https://gist.github.com/gfoidl/346ea7472f06e2024b8bdc9c598e547e#gistcomment-2773411) on that problem.

Reply:
That's a very interesting problem, but not really relevant here: this array is only allocated once in a constructor and never modified.

Reply:
>  only allocated once in a constructor and never modified.

👍🏻 super (thought I saw a write, but I was wrong).